request-plan9.go

```diff
@@ -16,7 +16,15 @@ func testOsSys(sys interface{}) error {
 	return nil
 }
 
-func toLocalPath(p string) string {
+func toLocalPath(workDir, p string) string {
+	if workDir != "" {
+		if !filepath.IsAbs(p) && !path.IsAbs(p) {
```

Comment:
I’m not sure why you’re using `filepath.IsAbs(p)` here. Paths coming in over the wire should be conformant with `path` only.

That’s why we use `filepath.FromSlash(p)` below.

Reply:
I was working around an issue where I was getting the work dir prefixed multiple times or things like `C:\C:\`. Ultimately some of that was probably a bug I had in the original implementation, but reading the code made me unsure if I could trust the input.

For instance, correct me if I'm wrong but this behavior doesn't conform to that guarantee:

Path is converted here:

https://github.com/pkg/sftp/blob/88c72168ab7facc5aca8de2be92d458c9153a2e0/server.go#L259

Set here:

https://github.com/pkg/sftp/blob/88c72168ab7facc5aca8de2be92d458c9153a2e0/server.go#L269

And a local path passed to `toLocalPath` here:

https://github.com/pkg/sftp/blob/88c72168ab7facc5aca8de2be92d458c9153a2e0/server.go#L449

Reply:
```
tempFile := filepath.Join(t.TempDir(), "sftp") // C:\Path\To\Tmp\sftp
file, err := client.Create(tempFile)
```
I cannot find this code anywhere in the existing source code.

Such usage is fundamentally erroneous. However, the behavior of this being turned into the local path of `C:\Path\To\Workdir\C:\Path\To\Tmp\sftp` is however the correct behavior, which should lead to an error on Windows because paths cannot contain `:` and a non error in Linux because `/path/to/workdir/C:/Path/To/Temp/sftp` is valid.

Building a temp directory appropriate for the local client simply does not make any sense in the aspect of making a file on the remote server.

Reply:
Ah yes. This was a similar thing for the posix versions as well, where a test can assume that `/tmp/asdf` is the same remote as it is local, and then it breaks when using the in memory handler. 🙃